src/Web/Rsp.IrasPortal.Web/Validators/QuestionSetValidator.cs

```diff
@@ -1,16 +1,45 @@
 ﻿using System.Diagnostics.CodeAnalysis;
 using FluentValidation;
+using FluentValidation.Results;
 using Rsp.IrasPortal.Web.Models;
 
 namespace Rsp.IrasPortal.Web.Validators;
 
 [ExcludeFromCodeCoverage]
 public class QuestionSetValidator : AbstractValidator<QuestionnaireViewModel>
 {
+    private bool includeMandatoryCheck;
+
+    protected override bool PreValidate(ValidationContext<QuestionnaireViewModel> context, ValidationResult result)
+    {
+        // Check if RootContextData contains the key and it's a bool
+        if (context.RootContextData.TryGetValue("ValidateMandatoryOnly", out var flag) && flag is bool value)
+        {
+            includeMandatoryCheck = value;
+        }
+        else
+        {
+            includeMandatoryCheck = false;
+        }
```

Comment:
You could directly use `includeMandatoryCheck`
```suggestion
        context.RootContextData.TryGetValue("ValidateMandatoryOnly", out bool includeMandatoryCheck);
```